runtime/stdlib/builtin.go

```diff
@@ -138,3 +140,113 @@ var LogFunction = NewStandardLibraryFunction(
 var HelperFunctions = StandardLibraryFunctions{
 	LogFunction,
 }
+
+var CreatePublicKeyFunction = NewStandardLibraryFunction(
+	sema.PublicKeyTypeName,
+	&sema.FunctionType{
+		Parameters: []*sema.Parameter{
+			{
+				Label:          sema.PublicKeyPublicKeyField,
+				Identifier:     sema.PublicKeyPublicKeyField,
+				TypeAnnotation: sema.NewTypeAnnotation(&sema.VariableSizedType{Type: &sema.UInt8Type{}}),
+			},
+			{
+				Label:          sema.PublicKeySignAlgoField,
+				Identifier:     sema.PublicKeySignAlgoField,
+				TypeAnnotation: sema.NewTypeAnnotation(sema.SignatureAlgorithmType),
+			},
+		},
+		ReturnTypeAnnotation:  sema.NewTypeAnnotation(sema.PublicKeyType),
+		RequiredArgumentCount: sema.RequiredArgumentCount(2),
+	},
+
+	func(invocation interpreter.Invocation) trampoline.Trampoline {
+		publicKey := invocation.Arguments[0].(*interpreter.ArrayValue)
+		signAlgo := invocation.Arguments[1].(*interpreter.BuiltinCompositeValue)
+
+		value := interpreter.NewPublicKeyValue(publicKey, signAlgo)
+		return trampoline.Done{Result: value}
+	},
+)
+
+// BuiltinValues
+
+var BuiltinValues = StandardLibraryValues{
+	SignatureAlgorithmValue,
+	HashAlgorithmValue,
+}
+
+var SignatureAlgorithmValue = StandardLibraryValue{
+	Name:  sema.SignatureAlgorithmTypeName,
+	Type:  getEnumType(sema.SignatureAlgorithmType, sema.SignatureAlgorithms),
+	Value: getEnumValue(sema.SignatureAlgorithmType, sema.SignatureAlgorithms),
+	Kind:  common.DeclarationKindEnum,
+}
+
+var HashAlgorithmValue = StandardLibraryValue{
+	Name:  sema.HashAlgorithmTypeName,
+	Type:  getEnumType(sema.HashAlgorithmType, sema.HashAlgorithms),
+	Value: getEnumValue(sema.HashAlgorithmType, sema.HashAlgorithms),
+	Kind:  common.DeclarationKindEnum,
+}
+
+func getEnumType(enumType *sema.BuiltinCompositeType, enumCases []sema.BuiltinEnumCase) *sema.SpecialFunctionType {
+	members := make([]*sema.Member, len(enumCases))
+	for _, algo := range enumCases {
+		members[algo.RawValue()] = sema.NewPublicEnumCaseMember(
+			enumType,
+			algo.Name(),
+			algo.DocString(),
+		)
+	}
+
+	constructorType := &sema.SpecialFunctionType{
+		FunctionType: &sema.FunctionType{
+			Parameters: []*sema.Parameter{
+				{
+					Identifier:     sema.EnumRawValueFieldName,
+					TypeAnnotation: sema.NewTypeAnnotation(enumType.EnumRawType),
+				},
+			},
+			ReturnTypeAnnotation: sema.NewTypeAnnotation(
+				&sema.OptionalType{
+					Type: enumType,
+				},
+			),
+		},
+		Members: sema.GetMembersAsMap(members),
+	}
+
+	return constructorType
+}
+
+func getEnumValue(enumType *sema.BuiltinCompositeType, enumCases []sema.BuiltinEnumCase) (value interpreter.Value) {
```

Comment:
Maybe make it clearer that this function only applies to built-in composite types with built-in enum cases